Marlin/planner.cpp

```diff
@@ -1212,6 +1219,23 @@ void Planner::set_position_mm(ARG_X, ARG_Y, ARG_Z, const float &e) {
   memset(previous_speed, 0, sizeof(previous_speed));
 }
 
+void Planner::set_position_mm_kinematic(const float position[NUM_AXIS]) {
+  const float* t=position;
+  #if PLANNER_LEVELING
+    float leveled[NUM_AXIS];
+    memcpy(leveled, position, sizeof(leveled));
+    apply_leveling(leveled);
+    t=leveled;
+  #endif
```

Comment:
Since `t` wants to be `const` in the normal case, better code here would be:
```cpp
  #if PLANNER_LEVELING
    float t[NUM_AXIS];
    memcpy(t, position, sizeof(t));
    apply_leveling(t);
  #else
    const float *t = position;
  #endif
```